lib/ddtrace/workers/trace_writer.rb

```diff
@@ -194,6 +196,22 @@ def write(trace)
         # Queue the trace if running asynchronously, otherwise short-circuit and write it directly.
         async? ? enqueue(trace) : write_traces([trace])
       end
+
+      private
+
+      def record_metrics
```

Comment:
In spirit of the `measure` suggestion I had, I think it might make sense to have this `record_metrics` take a block, then be used to wrap `perform#super`. Basically something like:

```
 - return yield if health metrics disabled
 - else
  - measure yield with appropriate clock
  - send health metric
  - ensure return yield result
```